sw/device/lib/runtime/pmp.h

```diff
@@ -0,0 +1,312 @@
+  kPmpRegionConfigureNapotError = kPmpRegionConfigureError,
+  kPmpRegionConfigureNapotBadArg = kPmpRegionConfigureBadArg,
+  kPmpRegionConfigureNapotBadRegion = kPmpRegionConfigureBadRegion,
+  kPmpRegionConfigureNapotBadAddress = kPmpRegionConfigureBadAddress,
+  kPmpRegionConfigureNapotWarlError = kPmpRegionConfigureWarlError,
+  /**
+   * NAPOT size must be at least 8bytes and Power Of Two. NAPOT address
+   * must be aligned to the size. Please see:
+   * "Volume II: RISC-V Privileged Architectures V20190608-Priv-MSU-Ratified",
+   * "3.6 Physical Memory Protection", "Table 3.10" and section
+   * "Address Matching".
+   */
+  kPmpRegionConfigureNapotBadSize,
+} pmp_region_configure_napot_result_t;
+
+/**
+ * Configures PMP address matching to Naturally aligned power-of-two (NAPOT).
+ *
+ * Size ≥ 8 bytes. When pmpcfgN is set to NAPOT then the protected region range
+ * is formed from a single encoded pmpaddrN address. NAPOT ranges make use of
+ * the low-order bits of the associated address register to encode the size of
+ * the range, please see the table "3.10" of the "Address Matching" section of
+ * the specification mentioned above.
+ *
+ * Example: NAPOT encoded address of yyyy011 describes the
+ * yyyy00000 .. yyyy11111 range. The index of the first "0", determines the
+ * range size ( 2^(3 + index) ).
+ *
+ * Please see the PMP specification reference and general information at the
+ * top of this header file.
+ *
+ * @param region PMP region to configure and set the address for.
+ * @param config PMP region configuration information.
+ * @param address PMP address to be set (must be aligned to `size`).
+ * @param size PMP protected address range size (must be power-of-two).
+ * @return `pmp_region_configure_napot_result_t`.
+ */
+PMP_WARN_UNUSED_RESULT
+pmp_region_configure_napot_result_t pmp_region_configure_napot(
+    pmp_region_index_t region, const pmp_region_config_t *config,
+    pmp_region_address_t address, uint32_t size);
+
+/**
+ * Configures PMP address matching to Top Of the Range (TOR).
+ *
+ * When pmpcfgN is set to TOR, the protected address range is formed from the
+ * pmpaddrN-1 and pmpaddrN registers. Note that the combination of pmpcfgN set
+ * to TOR and pmpcfgN-1 set to NAPOT does not form any meaningful range for TOR
+ * addressing mode. pmpcfgN TOR mode will take the bottom of the range
+ * pmpaddrN-1 address as is, without decoding the NAPOT address.
+ *
+ * Please see the PMP specification reference and general information at the
+ * top of this header file.
+ *
+ * @param region PMP region to configure and set the address for.
+ * @param config PMP region configuration information.
+ * @param address_start PMP bottom of the range address to be set. It must be
+ *        0 for the "0" region.
+ * @param address_end PMP top of the range address to be set.
+ * @return `pmp_region_configure_result_t`.
+ */
+PMP_WARN_UNUSED_RESULT
+pmp_region_configure_result_t pmp_region_configure_tor(
```

Comment:
Not sure whether I should match napot signature here (instead of 2 addresses, have address and size).

Reply:
Two addresses is fine for the moment :)